srv/modules/runners/replace.py

```
@@ -164,16 +168,28 @@ def minion_profile(minion, osds, grains):
     if yaml_file in files:
         for filename in files[yaml_file]:
             if os.path.exists(filename):
-                try:
-                    print("Renaming minion {} profile".format(minion))
-                    os.rename(filename, "{}-replace".format(filename))
-                    _insert_replace_flag(
-                        grains, disks, minion, osds, "{}-replace".format(filename)
-                    )
-                # pylint: disable=bare-except
-                except:
-                    log.error("Failed to rename minion {} profile".format(minion))
-                    os.rename("{}-replace".format(filename), filename)
+                if context == 'replace':
+                    try:
+                        print("Renaming minion {} profile".format(minion))
+                        os.rename(filename, "{}-replace".format(filename))
+                        _insert_replace_flag(
+                            grains, disks, minion, osds, "{}-replace".format(filename)
+                        )
+                    # pylint: disable=bare-except
+                    except:
+                        log.error("Failed to rename minion {} profile".format(minion))
+                        os.rename("{}-replace".format(filename), filename)
```

This seem strange. We revert the rename, if the rename has failed? Maybe we do expect multiple exceptions here?

---

I  can give some context (as the original author):

The log message is a bit misleading, that is right. Two things can fail: renaming the file or inserting the replace flags. In each case, the file is brought back to the state before it got renamed and a "generic" error message is provided. This can definitely be improved.

The bare except is here because I am not quite sure what exceptions we are looking for (probably IO and dict look up related)

---

This was initially intended to act as sort of a 'rollback' mechanism for the `_insert_replace_flag` method. I think this can stay in there for now as it should not cause any harm. If any of the above operations fail, we just end up with what we had.

---

Well I'm wondering...if the `os.rename(src, dest)` fails (OSError if `dest` exists) we then take `dest` and try to rename it to `src`. This will throw another exception since `src` exists. Granted this shouldn't happen™ (or is somewhat unlikely) but it defeats the purpose of the exception handling. Have I missed anything?
If not I'd vote for proper exception handling here. The rollback-rename should only happen for `KeyError` coming from `_insert_replace_flag`. OSError should log a message but probably rethrow the exception (or is there a sensible way to handle it and continue?). The potential IOError (from `open` in `_insert_replace_flag`) should bubble up imho.

---

Why not `except KeyError` then (and maybe YAMLError in case the dump can gor wrong?)? If nothing else it makes the intent and assumptions obvious to the next person.
I will concede this can be a code hygiene fix in another PR.

---

```python
try:
   ...
except FirstException:
   handle_first_one()

except SecondException:
   handle_second_one()
```

I wasn't aware that this is even possible :>

I'll give in if we move this to a separate PR.